src/utils/opt_in_checker.py

```diff
@@ -0,0 +1,260 @@
+                      "consent to the collection of this data, a \"N\" will prevent the anonymous data from " \
+                      "being sent back to Intel. " \
+                      "Thanks for replying with a \"Y\"!".format(opt_in_out_script_run_command)
+    opt_in_question = "Intel would like your permission to collect software performance and usage data for the " \
+                      "purpose of improving Intel products and services. This data will be collected directly " \
+                      "by Intel or through the use of Google Analytics. This data will be stored in countries " \
+                      "where Intel or Google operate.\nYou can opt-out at any time in the future by running {}.\n" \
+                      "More Information is available at {}.\n" \
+                      "Please type ‘Y’ to give your consent or ‘N’ to decline.".format(opt_in_out_script_run_command,
+                                                                                       doc_link)
+    opt_in_question_incorrect_input = "Please type ‘Y’ to give your consent or ‘N’ to decline."
+    response_confirmation_accept = "The selected option was to collect telemetry data."
+    response_confirmation_decline = "The selected option was NOT to collect telemetry data."
+    response_confirmation_timer_reached = "The timer has expired and no data will be collected."
+
+    @staticmethod
+    def _ask_opt_in(question: str, timeout: int):
+        """
+        Runs input with timeout and checks user input.
+        :param question: question that will be printed on the screen.
+        :param timeout: timeout to wait.
+        :return: opt-in dialog result.
+        """
+        colored_print(question)
+        answer = input_with_timeout(prompt='>>', timeout=timeout)
+        answer = answer.lower()
+        if answer == "n" or answer == "no":
+            colored_print(OptInChecker.response_confirmation_decline)
+            return DialogResult.DECLINED
+        if answer == "y" or answer == "yes":
+            colored_print(OptInChecker.response_confirmation_accept)
+            return DialogResult.ACCEPTED
+        return DialogResult.TIMEOUT_REACHED
+
+    def opt_in_dialog(self):
+        """
+        Runs opt-in dialog until the timeout is expired.
+        :return: opt-in dialog result.
+        """
+        start_time = time.time()
+        answer = self._ask_opt_in(self.opt_in_question, self.dialog_timeout)
+        time_passed = time.time() - start_time
+        while time_passed < self.dialog_timeout and answer == DialogResult.TIMEOUT_REACHED:
+            answer = self._ask_opt_in(self.opt_in_question_incorrect_input, self.dialog_timeout - time_passed)
+            time_passed = time.time() - start_time
+
+        if answer == DialogResult.TIMEOUT_REACHED:
+            colored_print(OptInChecker.response_confirmation_timer_reached)
+
+        return answer
```

Comment:
Could you please explain how this opt dialog is integrated into the MO, POT tool? Now it is not clear what is telemetry external API.

Reply:
Opt-in dialog is run from Telemetry class constructor. So telemetry initialization runs it. 
For example in MO here we initialize Telemetry: https://github.com/openvinotoolkit/openvino/blob/43c45d3065f98dac29fb36885ff9a5bc2691143d/tools/mo/openvino/tools/mo/main.py#L437

This line will run opt-in dialog. Same for other components.